alvr/server_openvr/src/lib.rs

```diff
@@ -414,30 +408,61 @@ pub extern "C" fn shutdown_driver() {
     SERVER_CORE_CONTEXT.write().take();
 }
 
+// Check that there is no active dashboard instance not part of this driver installation
+pub fn should_initialize_driver(driver_layout: &afs::Layout) -> bool {
+    if let Some(proc) = sysinfo::System::new_all()
+        .processes_by_name(OsStr::new(&afs::dashboard_fname()))
+        .next()
+    {
+        if let Some(dashboard_path) = proc.exe() {
+            if let Some(other_fs_layout) =
+                alvr_filesystem::filesystem_layout_from_dashboard_exe(dashboard_path)
+            {
+                return driver_layout == &other_fs_layout;
+            }
+        }
+    }
+
+    true
+}
+
 /// This is the SteamVR/OpenVR entry point
 /// # Safety
 #[no_mangle]
 pub unsafe extern "C" fn HmdDriverFactory(
     interface_name: *const c_char,
     return_code: *mut i32,
 ) -> *mut c_void {
+    let Ok(driver_dir) = alvr_server_io::get_driver_dir_from_registered() else {
+        return ptr::null_mut();
+    };
+    let Some(filesystem_layout) =
+        alvr_filesystem::filesystem_layout_from_openvr_driver_root_dir(&driver_dir)
+    else {
+        return ptr::null_mut();
+    };
+
+    if !should_initialize_driver(&filesystem_layout) {
+        return ptr::null_mut();
+    }
```

Comment:
I feel like these failing totally silently might be pretty bad, is there something we could do to try to get some logging? Or do we just hope these never happen and if they do they fix themselves with a restart?

Reply:
First two let else are required to even setup logging. but before they were simply `unwrap`ped so it's relatively safe

Reply:
In the end I decided also `should_initialize_driver` can stay before the logging initialization. There is not much code in there and it's non-failing

Reply:
Well yea but it's a check intended to bail out, a notice there might be good... Otoh it's fixed by simply closing all alvr instances and restarting so probably not too bad

Reply:
Ah I just realized this would happen for all registered drivers and thus if you have multiple you might get multiple logs even tho the failure is actually wanted, so no logging is probably best.